h/services/user_signup.py

```diff
@@ -80,7 +85,21 @@ def signup(self, require_activation=True, **kwargs):
 
         # Create a new activation for the user
         if require_activation:
-            self._require_activation(user)
+            try:
+                self._require_activation(user)
+            except IntegrityError as e:
+                if (
+                    'duplicate key value violates unique constraint "uq__user__email"'
+                    in e.args[0]
+                ):
+                    log.warning("email conflict error occured during user signup")
+                    raise ConflictError(
+                        "The email account {} has already been registered.".format(
```

Comment:
Strings which are expected to appear in the UI should be marked as translatable using `_` (imported via `from h.i18n import TranslationString as _` ).